### PR TITLE
Connect-api

### PR DESCRIPTION
**Pull request by davidigbo**
In this pull request, I implement api for the bookstore application by following these steps:

-Make use of the [Bookstore API] 
-Use [axios](https://axios-http.com/docs/intro) with [createAsyncThunk](https://redux-toolkit.js.org/api/createAsyncThunk) to make your HTTP requests to the backend
-Fetch your books from the Bookstore API after the component has rendered
-Refactor your add book and remove book features to persist your changes in the server
-Your application should behave the same way after the refactoring

### CODE REVIEW

File: src/components/bookItems.js [hunk 1]
```diff
@@ -1,28 +1,52 @@
...
-        Remove Book
-      </button>
+    <div className="book-wrapper">
+      <div className="book-datail">
+        <div className="book-category">{JSON.parse(book).category}</div>
+        <div className="book-title">{JSON.parse(book).title}</div>
+        <div className="book-author">{JSON.parse(book).author}</div>
+        <div className="bool-detail-btn">
+          <button type="button" className="book-comment">Comment</button>
+          <button
```
Comment (Qoosim) [inline src/components/bookItems.js]: - [ ] The book prop you passed into the BookItem component is seen as being undefined. As a result there is an error that's preventing the app from loading on the browser. Kindly take a look and fix.

### Error on the browser
error

**Review by Qoosim — CHANGES_REQUESTED** [review]
#  *Status: Required Changes* :recycle:

Hi , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct . :heavy_check_mark: 




## Required Changes ♻️
_Check the comments under the review._

- [ ] Kindly fix the MIT License link on your README file. It is broken at this moment.

license


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by davidigbo** [comment]
Thanks for oyu reviewing my code, I will implement the changes

File: README.md [hunk 2]
```diff
@@ -129,12 +129,8 @@ I would like say to thank Microverse
...
 <p align="right">(<a href="#readme-top">back to top</a>)</p>
 
-<!-- LICENSE -->
-
 ## 📝 License <a name="license"></a>
 
-This project is [MIT](./LICENSE) licensed.
-
-_NOTE: we recommend using the [MIT license](https://choosealicense.com/licenses/mit/) - you can set it up quickly by [using templates available on GitHub](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-a-license-to-a-repository). You can also use [any other license](https://choosealicense.com/licenses/) if you wish._
+This project is [MIT](./LICENSE.md) licensed.
```
Comment (Zuhaib042) [inline README.md]: - [ ] As suggested by the previous reviewer, this link is still broken. The reason of this is that you don't have `LICENSE.md` file in your repo. kindly consider adding a LICENSE.md in your repo and then check if the link works before submitting it again.

File: src/components/bookItems.js [hunk 3]
```diff
@@ -1,28 +1,103 @@
...
-      <button
-        type="button"
-        className="book-remove"
-        id={JSON.parse(book).id}
-        onClick={(evet) => dispatch(removeBook(evet.target.id))}
-      >
-        Remove Book
-      </button>
+    <div className="book-wrap">
+      <div className="book-detail">
```
Comment (Zuhaib042) [inline src/components/bookItems.js]: - [ ] As suggested by the previous reviewer, The `book` prop in this `BookItem` component is `undefined`. This error is popping up in the browser so the app is not working. kindly consider fixing this bug before submitting it again.
the error in the browser
qw

File: src/components/bookItems.js [hunk 4]
```diff
@@ -1,28 +1,103 @@
...
+//       </div>
+//     </div>
+//   );
+// };
+// BookItem.propTypes = {
+//   book: PropTypes.string.isRequired,
+//   id: PropTypes.string.isRequired,
+// };
+
+// export default BookItem;
```
Comment (Zuhaib042) [inline src/components/bookItems.js]: - [OPTIONAL] Kindly consider removing this commented code as it makes our codebase messy and unprofessional.

**Review by Zuhaib042 — CHANGES_REQUESTED** [review]
Hi @david30-maker  

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 2 above)
Comment (julie-ify) [inline README.md]: - [ ] Kindly fix this issue of missing MIT link.

  image

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @david30-maker,

Good job so far!

## Highlights!
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by julie-ify — COMMENTED** [review]
Hi @david30-maker,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

### Kindly fix the issue highlighted earlier (or push the changes). The MIT link on your readme file is still broken.

image

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by Davidosky007 — COMMENTED** [review]
Hi @david30-maker 👋, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has a broken MIT link kindly fix this 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by TadesseDev — APPROVED** [review]
### Hi @david30-maker 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
giphy (1)



## Highlights
 - [x] Good job making the changes requested by the previous code reviewer. ✅

## Optional suggestions
- [Optional] Please consider removing non-important contents from your readme, The template is designed by Microverse for us all to use to document our projects professionally and has it filled with content related to each project we may be working on. currently, I can see you have not removed the guiding contents  which start with `>` and `directions set by Microverser`

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by davidigbo** [comment]
Thanks, @TadesseDev for reviewing my code

**Pull request merged**